File: Java/LC202/src/Solution.java
public class Solution {
    public boolean isHappy(int n) {
        StringBuilder sb = new StringBuilder().append(n);

        while (sb.length() > 1) {
            int sos = 0;
            for (int i = 0; i < sb.length(); i++) {
                sos += Math.pow(sb.charAt(i) - '0', 2);
            }
            sb = new StringBuilder().append(sos);
        }

        String s = sb.toString();

        return s.equals("1") || s.equals("7");
    }
}

File: Kotlin/LC202/src/main/kotlin/Solution.kt
class Solution {
    fun isHappy(n: Int): Boolean {
        var sb: StringBuilder = StringBuilder().append(n)

        while (sb.length > 1) {
            var sos = 0
            for (i in sb.indices) {
                sos += (sb[i] - '0') * (sb[i] - '0')
            }
            sb = StringBuilder().append(sos)
        }

        val s: String = sb.toString()
        return s == "1" || s == "7"
    }
}